### PR TITLE
Preliminary refactoring for the event based from GMFs calculator

### PR DESCRIPTION
**Pull request by micheles**
The big change is in the way the GMFs are stored. Now there is a dataset for each hazard site, containing all GMVs for all intensity measure types, all realizations and all event ids.

Requires the companion https://github.com/gem/oq-hazardlib/pull/487
The tests are running here: https://ci.openquake.org/job/zdevel_oq-engine/2170

### CODE REVIEW

File: openquake/risklib/riskinput.py [hunk 1]
```diff
@@ -456,11 +462,13 @@ def get_hazard(self, rlzs_assoc, monitor=Monitor()):
...
         """
+        if rlzs_assoc is None:  # case ebr_gmf
+            return self.hazard_by_site
         return [{imt: rlzs_assoc.combine(haz[imt]) for imt in haz}
                 for haz in self.hazard_by_site]
 
     def __repr__(self):
-        return '<%s taxonomies=%s, weight=%d>' % (
+        return '<%s IMT=%s, taxonomy=%s, %d asset(s)>' % (
             self.__class__.__name__, ', '.join(self.taxonomies), self.weight)
```
Comment (ptormene) [inline openquake/risklib/riskinput.py]: Missing argument for IMT

File: openquake/commonlib/export/hazard.py [hunk 2]
```diff
@@ -560,7 +565,21 @@ def export_gmf(ekey, dstore):
         logging.warn(GMF_WARNING, dstore.hdf5path)
     fnames = []
     for rlz in rlzs_assoc.realizations:
-        gmf_arr = gmf_data['%04d' % rlz.ordinal].value
+        if n_gmfs:
+            # TODO: change to use the prefix rlz-
```
Comment (ptormene) [inline openquake/commonlib/export/hazard.py]: Please check if this has to be done in this PR or in a separate one

File: openquake/calculators/ucerf_classical.py [hunk 3]
```diff
@@ -300,7 +300,7 @@ def ucerf_classical_hazard_by_rupture_set(
             monitor=monitor)
 
     else:
-        dic[src_group_id] = ProbabilityMap()
+            dic[src_group_id] = ProbabilityMap(len(imtls.array), len(gsims))
```
Comment (ptormene) [inline openquake/calculators/ucerf_classical.py]: Please check the indentation

**Comment by ptormene** [comment]
Apart from minor comments, LGTM

**Comment by micheles** [comment]
Addressed the comments

**Pull request merged**